plugins/package-managers/gradle-plugin/src/main/kotlin/OrtModelBuilder.kt

```diff
@@ -0,0 +1,227 @@
+    private val visitedProjects = mutableSetOf<ModuleVersionIdentifier>()
+
+    // Note: Using "LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE" does not work for the lookup as it is not String-typed.
+    private val libraryElementsAttribute = Attribute.of("org.gradle.libraryelements", String::class.java)
+
+    private val logger = Logging.getLogger(OrtModelBuilder::class.java)
+    private val errors = mutableListOf<String>()
+    private val warnings = mutableListOf<String>()
+
+    override fun canBuild(modelName: String): Boolean =
+        modelName == OrtDependencyTreeModel::class.java.name
+
+    override fun buildAll(modelName: String, project: Project): OrtDependencyTreeModel {
+        repositories = project.repositories.associate { it.name to (it as? UrlArtifactRepository)?.url?.toString() }
+
+        val resolvableConfigurations = project.configurations.filter { it.isResolvable() }
+
+        val ortConfigurations = resolvableConfigurations.mapNotNull { config ->
+            // Get the root of the resolved dependency graph. This is also what Gradle's own "dependencies" task uses to
+            // recursively obtain information about resolved dependencies. Resolving dependencies triggers the download
+            // of metadata (like Maven POMs) only, not of binary artifacts, also see [1]. However, Gradle stores the
+            // metadata e.g. for Maven parent POMs in its own binary "descriptor.bin" format only, as long as the parent
+            // POM artifact is no explicitly requested.
+            //
+            // [1]: https://docs.gradle.org/current/userguide/dependency_management.html#obtaining_module_metadata
+            val root = config.incoming.resolutionResult.root
+
+            // Omit configurations without dependencies.
+            root.dependencies.takeUnless { it.isEmpty() }?.let { dep ->
+                // Reset visited dependencies and projects per configuration.
+                visitedDependencies.clear()
+                visitedProjects.clear()
+
+                OrtConfigurationImpl(name = config.name, dependencies = dep.toOrtDependencies())
+            }
+        }
+
+        return OrtDependencyTreeModelImpl(
+            group = project.group.toString(),
+            name = project.name,
+            version = project.version.toString(),
+            configurations = ortConfigurations,
+            repositories = repositories.values.filterNotNull(),
+            errors = errors,
+            warnings = warnings
+        )
+    }
+
+    private fun Configuration.isResolvable(): Boolean {
+        val canBeResolved = GradleVersion.current() < GradleVersion.version("3.3") || isCanBeResolved
+
+        val isDeprecatedConfiguration = GradleVersion.current() >= GradleVersion.version("6.0")
+                && this is DeprecatableConfiguration && resolutionAlternatives != null
+
+        return canBeResolved && !isDeprecatedConfiguration
```

Comment:
Am I right with assuming that a deprecated configuration is not resolvable?

Reply:
No, deprecated configurations *are* resolvable. That's why they need to be filtered out explicitly.

Reply:
I got this hint from the function name, do you agree function name is mis-leading?

Reply:
Yes, I can see how to miss the notion between "canBeResolved" and "isResolvable". I'm open for proposals for better names for the extension function. Maybe "shouldResolve", "isAccepted" or "isEligible"?

Reply:
I think I'll go for "isRelevent" for now, also in context of a minor refactoring.

Reply:
Hm, my only idea is `shouldBeIgnored()` which I'm not sure if I like.  
Maybe separate functions could do. `project.configurations.filter { it.isResolvable() && !it.isDeprecated() }`

I believe `shouldResolve()` would make it clearer to me compared to the current code.

Reply:
Maybe while at it, a comment could be added explaining _why_ deprecated configurations are ignored.

Reply:
> Isn't that obvious? Because the are deprecated (in favor of another configuration). 

Does the deprecation imply that the build script is not using that configuration to define it's dependencies?
Or could the build script still use deprecated configs?